runtime/parachains/src/disputes/migration.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Storage migration(s) related to disputes pallet
+
+use frame_support::traits::StorageVersion;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+pub mod v1 {
+	use super::*;
+	use crate::disputes::{Config, Pallet};
+	use frame_support::{
+		pallet_prelude::*, storage_alias, traits::OnRuntimeUpgrade, weights::Weight,
+	};
+	use primitives::v2::SessionIndex;
+	use sp_std::prelude::*;
+
+	#[storage_alias]
```

Comment:
Out of curiosity: What is the difference of `storage_alias` as compared to `pallet::storage`? I could not find any documentation in the substrate source.

Reply:
My very naive explanation is that `pallet::storage` is intended to be used inside pallet. E.g. if I try to use `#[pallet::storage]` here I get bunch of compilation errors.

`storage_alias` seems to generate similar code for working with the storage without the need of the pallet itself. It's documentation says `Useful for creating a storage-like struct for test and migrations.`

Why I use `storage_alias` here? I saw it in the PR Kian shared: https://github.com/paritytech/substrate/pull/12230/files#diff-c69ee0dc305217a4f0fac4506e7d9f3535814a07b49c62f40cdc7991b7ea3114

I'd also appreciate a better explanation on the difference between `storage_alias` and `pallet::storage` though.